### PR TITLE
Add alternate timestamp functionality

### PR DESCRIPTION
**Pull request by goatzillax**
Two commits add the config options and the ability to use postion_ms from the VideoStream source for event timestamps.

### CODE REVIEW

File: dvr_scan/video_joiner.py [hunk 1]
```diff
@@ -84,8 +85,20 @@ def position(self) -> FrameTimecode:
...
+        return self._cap.position_ms
+
+    @property
+    def last_position_ms(self) -> float:
+        return self._last_position_ms
+
     def read(self, decode: bool = True) -> Optional[numpy.ndarray]:
         """Read/decode the next frame."""
+
+        #  I get the feeling when you run off the end of a file the position_ms returns 0.
```
Comment (Breakthrough) [inline dvr_scan/video_joiner.py:99]: Could we add a unit test to verify this?  You can [add one to `tests/test_video_joiner.py`](https://github.com/Breakthrough/DVR-Scan/blob/develop/tests/test_video_joiner.py).

**Comment by Breakthrough** [comment]
This is awesome, thanks so much!  Could you add a few short unit tests and fix the code formatting checks to land this?

(can autoformat files by running `python -m yapf -i -r dvr_scan/ tests/` from the root of the repo)

**Comment by goatzillax** [comment]
I reverted the workaround with the last_position_ms since that wasn't the root cause, so accordingly there's no testcase for it.

**Comment by Breakthrough** [comment]
Thanks for fixing this up!  I can add a test after merging if you don't think it needs one.  What happened to be the issue?

**Review by Breakthrough — APPROVED** [review]

**Comment by goatzillax** [comment]
> Thanks for fixing this up! I can add a test after merging if you don't think it needs one. What happened to be the issue?

The root issue was I forgot the decoding was happening asynchronously on its own thread, so calling the position on the underlying VideoCapture element in (different) scan thread would be out of sync with the frame being worked on.  Most of the time it was subtle until the end of the file where it was not so subtle.

I can add that test; seems reasonable.

**Comment by goatzillax** [comment]
I added the test and adjusted a few things.

For a motion event that's still going when a file ends, I was adding the the post_even_len since ffmpeg doesn't seem to care if you do that, and that seemed more inclusive to me, but I've changed it to match the existing behavior.

The other thing is the start events are slightly off because frame_num seems to be shifted to after the frame.

Doing something similar for the PTS case might be counterproductive somewhere down the line.  I added the testcase and gave it a 1 frame tolerance to compensate.

**Review by Breakthrough — APPROVED** [review]
Thanks for fixing this up.  Really appreciate the PR!

Will make sure this gets in for the next release.

**Pull request merged**